src/mmap.rs

```diff
@@ -64,12 +71,14 @@ pub struct GuestRegionMmap {
 
 impl GuestRegionMmap {
     /// Create a new memory-mapped memory region for guest's physical memory.
-    /// Note: caller needs to ensure that (mapping.len() + guest_base) doesn't wrapping around.
-    pub fn new(mapping: MmapRegion, guest_base: GuestAddress) -> Self {
-        GuestRegionMmap {
+    pub fn new(mapping: MmapRegion, guest_base: GuestAddress) -> result::Result<Self, Error> {
+        if guest_base.0.checked_add(mapping.len() as u64).is_none() {
+            return Err(Error::InvalidGuestRegion);
+        }
```

Comment:
Should we also check that the offset and len are page aligned? Or just let the mmap syscall to check them?

Reply:
I guess it's better to check it ourselves since we cannot assume that both Linux and Windows `mmap` syscall will check for that, right?